### ISSUE TITLE
How to update bn parameters?

### ISSUE THREAD

**Issue by gsygsy96** [CVLAB-Unibo/Learning2AdaptForStereo#5]
I wonder how to update bn parameters during the adaptation stage, since the batchsize is 1.

**Comment by AlessioTonioni** [comment]
The DispNet that was used for the tests do not have batch norms. In general during adaptation you can keep the mean and variance freezed or accumulate statistics over multiple frames and then update means and vars.

**Comment by gsygsy96** [comment]
I see. Original Dispnet doesn't has batchnorm layer. Sorry for this mistake. As for your suggested update manner, I have some questions. If I accumulate the gradient, it sounds that the adaptation is changed to be an offline process?

**Comment by AlessioTonioni** [comment]
You would not accumulate the gradients but rather the statistics (mean and variance) used by the batch norm layers to normalize the inputs. 
Once you have processed enough samples and the statistics have "converged" you can replace them to the one learned on the training set. Some works also had more success using instance norm rather than batch norm since they will not suffer from this problem from the very beginning.

**Comment by gsygsy96** [comment]
I see... thanks for your support!
